### PR TITLE
CfW: fix MouseEvent to PointerButton mapping

### PR DESCRIPTION
**Pull request by eymar**
To run the tests:
- run them from IDEA
- or using an extra gradle property: `./gradlew :compose:ui:ui:cleanWasmJsBrowserTest :compose:ui:ui:wasmJsBrowserTest -Pjetbrains.cfw.tests.useChrome=true `

It tests the fix for this issue: https://github.com/JetBrains/compose-multiplatform/issues/4134

### CODE REVIEW

File: compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt [hunk 1]
```diff
@@ -18,9 +18,16 @@ package androidx.compose.ui.input.pointer
 
 import org.w3c.dom.events.MouseEvent
 
-internal val MouseEvent.composeButton get() =
+internal val MouseEvent.composeButton get(): PointerButton? {
     // https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button
```
Comment (MatkovIvan) [inline compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt]: It's a comment for `when (button`, please move it to that

File: compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt [hunk 2]
```diff
@@ -33,6 +40,7 @@ internal val MouseEvent.composeButton get() =
         4 -> PointerButton.Forward
         else -> PointerButton(-1)
     }
+}
```
Comment (MatkovIvan) [inline compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt:42]: If it's nullable now we can return null in `else` section

File: compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt [hunk 3]
```diff
@@ -18,9 +18,16 @@ package androidx.compose.ui.input.pointer
...
+internal val MouseEvent.composeButton get(): PointerButton? {
     // https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button
-    when (button.toInt()) {
+    // `MouseEvent.button` property only guarantees to indicate which buttons are pressed during
+    // events caused by pressing or releasing one or multiple buttons
+
+    val eventType = type
+    if (eventType != "mousedown" && eventType != "mouseup") {
+        return null
+    }
```
Comment (MatkovIvan) [inline compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt]: whe
```suggestion
    when (type) {
        "mousedown", "mouseup" -> return null
    }
```

File: compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt (hunk 2 above)
Comment (eymar) [reply compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt:42]: actually, there can be more buttons than listed here. Neither `-1` nor `null` are okay. 
I'll return `else -> PointerButton(button.toInt())` for now. When I have a mouse with more buttons, I'll check it again.  

__

wait a second. the tests fail now. :D let me check

File: compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt (hunk 1 above)
Comment (eymar) [reply compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt]: ack

File: compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt (hunk 3 above)
Comment (eymar) [reply compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt:29]: ack

Comment (eymar) [reply compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt]: This suggestion is not correct. It should return null for all other events except those two.

Comment (MatkovIvan) [reply compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt]: Ah, then
```kt
when (type) {
    "mousedown", "mouseup" -> {}
    else -> return null
}
```

File: compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt (hunk 2 above)
Comment (eymar) [reply compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt:42]: the test failed for another reason (fixed). The above comment remains valid from my point of view.

File: compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt (hunk 3 above)
Comment (eymar) [reply compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/input/pointer/PointerEvent.web.kt]: fixed

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**